collections/requirements.yml

```diff
@@ -0,0 +1,4 @@
+---
```

Comment:
I'm not entirely sure of adding this here as we don't have a `collections` dir so far. Maybe it would be better to only document this here: https://github.com/kubernetes-sigs/kubespray/blob/master/docs/ansible/ansible_collection.md WDYT?

Reply:
ansible-lint will look at this file when you simply run it in the top of the repo with only ansible-core installed. After commit https://github.com/kubernetes-sigs/kubespray/pull/11384/commits/cf56057650829ad0ec17d97f4fcea69075f002c0 The error **syntax-check[unknown-module]** was resolved.

Reply:
It also looks at the root dir it seems (https://github.com/ansible/ansible-lint/blob/288db8a90d640cea6a1b182ca0db22c89d06b49c/docs/usage.md?plain=1#L322) so would be preferable to be there IMO

Reply:
Actually we have some dependencies directly in galaxy.yml so the missing ones should just be added there

Reply:
I moved the requirements.yml to the root, and moved gluster.gluster in tests/requirements.yml.
This way we can simply run ansible-lint in the root with ansible-core, and that will pass now.

Reply:
The galaxy file is for the collection.

Reply:
I would prefer to use the galaxy.yml directly. I am not sure we should support the combination of not using ansible-core and not using the kubespray collection as well. This improvement is about modernizing kubespray in some sort and using kubespray via the collection is technically the modern way of using kubespray.

Right now we mandate users to install kubespray requirements via `requirements.txt` so what you are trying to do not go into this direction. I would be ok havinh this expressed via the kubespray collection/Galaxy file as there are already collection requirements although they are not complete. Also galaxy.yml is parsed by ansible-lint so even that use case is getting covered.

Reply:
Ok, I'll add them to `galaxy.yml.
BTW `requirements.txt` lists a yanked version of Ansible. I'll add a commit.